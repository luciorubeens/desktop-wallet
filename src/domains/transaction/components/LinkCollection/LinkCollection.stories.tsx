import React from "react";
import { useForm } from "react-hook-form";

import { LinkCollection } from "./LinkCollection";

export default { title: "Domains / Transaction / Components / LinkCollection" };

const types = [
	{
		label: "Facebook",
		value: "facebook",
	},
	{
		label: "Twitter",
		value: "twitter",
	},
	{
		label: "Instagram",
		value: "instagram",
	},
];

export const Default = () => {
	const { control, register } = useForm();

	return (
		<LinkCollection
			control={control}
			registerRef={register}
			selectOptions={types}
			name="socialMedia"
			title="Social Media"
			description="Tell people more about yourself through social media"
			itemLabel="media"
		/>
	);
};
