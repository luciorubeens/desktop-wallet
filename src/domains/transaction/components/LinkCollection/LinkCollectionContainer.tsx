import { Collapse } from "app/components/Collapse";
import { Icon } from "app/components/Icon";
import React, { useState } from "react";

const CollapseToggleIcon = ({ isExpanded }: { isExpanded?: boolean }) =>
	isExpanded ? (
		<Icon
			name="ChevronUp"
			width={10}
			height={10}
			className="flex items-center justify-center w-5 h-5 text-white rounded-full bg-theme-primary"
		/>
	) : (
		<Icon
			name="ChevronDown"
			width={10}
			height={10}
			className="flex items-center justify-center w-5 h-5 rounded-full text-theme-primary bg-theme-primary-contrast"
		/>
	);

type Props = {
	defaultIsExpanded?: false;
	title: string;
	description: string;
	children: React.ReactNode;
};

export const LinkCollectionContainer = ({ defaultIsExpanded, title, description, children }: Props) => {
	const [isExpanded, setIsExpanded] = useState<boolean>(defaultIsExpanded!);

	return (
		<div data-testid="LinkCollectionContainer">
			<div
				data-testid="LinkCollectionContainer__header"
				className="flex items-center justify-between cursor-pointer"
				onClick={() => setIsExpanded((prev) => !prev)}
			>
				<span className="text-lg font-semibold">{title}</span>

				<div data-testid="LinkCollectionContainer__collapse-toggle">
					<CollapseToggleIcon isExpanded={isExpanded} />
				</div>
			</div>

			<p data-testid="LinkCollectionContainer__description" className="mt-2 text-theme-neutral-dark">
				{description}
			</p>

			<Collapse data-testid="LinkCollectionContainer__collapse" isOpen={isExpanded}>
				<div className="mt-4">{children}</div>
			</Collapse>
		</div>
	);
};
