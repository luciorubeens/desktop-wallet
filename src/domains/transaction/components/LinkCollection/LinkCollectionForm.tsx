import { Button } from "app/components/Button";
import { Form, FormField, FormLabel } from "app/components/Form";
import { Input } from "app/components/Input";
import { Select } from "app/components/SelectDropdown";
import React from "react";
import { useForm } from "react-hook-form";
import { useTranslation } from "react-i18next";

import { EntityLink } from "./LinkCollection.models";

type Props = {
	selectOptions: { label: string; value: string }[];
	append: (link: EntityLink) => void;
	itemLabel: string;
};

export const LinkCollectionForm = ({ selectOptions, append, itemLabel }: Props) => {
	const { t } = useTranslation();

	// TODO: Validate URL
	const form = useForm();
	const { register, reset } = form;

	const handleSubmit = (link: EntityLink) => {
		reset();
		append(link);
	};

	return (
		<Form
			data-testid="LinkCollectionForm"
			context={form}
			onSubmit={(data: any) => handleSubmit(data as EntityLink)}
			className="grid col-gap-2 row-gap-4"
			style={{
				gridTemplateColumns: "40% 60%",
			}}
		>
			<FormField name="type">
				<FormLabel label={`Add ${itemLabel}`} />
				<Select options={selectOptions} ref={register({ required: true })} />
			</FormField>

			<FormField name="value">
				<FormLabel label={t("COMMON.LINK")} />
				<Input data-testid="LinkCollectionForm__value" ref={register({ required: true })} />
			</FormField>

			<Button data-testid="LinkCollectionForm__submit" className="col-span-2" variant="plain" type="submit">
				{t("TRANSACTION.ADD_LINK")}
			</Button>
		</Form>
	);
};
