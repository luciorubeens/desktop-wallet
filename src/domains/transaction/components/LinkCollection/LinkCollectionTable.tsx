import { Button } from "app/components/Button";
import { Icon } from "app/components/Icon";
import { RadioButton } from "app/components/RadioButton";
import { Table } from "app/components/Table";
import React from "react";
import { ValidationRules } from "react-hook-form";
import { useTranslation } from "react-i18next";

import { EntityLink } from "./LinkCollection.models";

type Props = {
	name: string;
	fields: EntityLink[];
	remove: (index: number) => void;
	registerRef: (options?: ValidationRules) => (ref: HTMLInputElement | null) => void;

	optionsTypes?: string[];
	optionsTypesTitle?: string;
	onOptionChange?: (link: EntityLink) => void;
	optionChecked?: number;
};

export const LinkCollectionTable = ({
	remove,
	optionsTypesTitle,
	optionsTypes,
	registerRef,
	fields,
	optionChecked,
	onOptionChange,
	name,
}: Props) => {
	const { t } = useTranslation();

	const columns = [];

	if (optionsTypesTitle) {
		columns.push({
			Header: optionsTypesTitle,
			accessor: "isSelected",
		});
	}

	columns.push(
		{
			Header: t("COMMON.NAME"),
			accessor: "type",
		},
		{
			Header: t("COMMON.LINK"),
			accessor: "link",
		},
		{
			Header: " ",
		},
	);

	return (
		<div data-testid="LinkCollectionTable">
			<Table columns={columns} data={fields}>
				{(rowData: EntityLink, rowIndex: any) => (
					<tr
						data-testid="LinkCollectionTable__row"
						key={rowData.value}
						className="font-semibold border-b border-theme-neutral-200"
					>
						{optionsTypesTitle && (
							<td className="w-16 text-center">
								{optionsTypes?.includes(rowData.type) && (
									<RadioButton
										data-testid="LinkCollectionTable__row__option"
										checked={optionChecked === rowIndex}
										onChange={() => onOptionChange?.(rowData)}
									/>
								)}
							</td>
						)}

						<td
							data-testid="LinkCollectionTable__row__type"
							className={`w-40 ${rowIndex > 0 ? "py-6" : "pb-6 pt-2"}`}
						>
							<input
								className="hidden"
								name={`${name}[${rowIndex}].type`}
								ref={registerRef()}
								defaultValue={rowData.type}
								readOnly
							/>
							<span>{t(`TRANSACTION.LINK_TYPES.${rowData.type.toUpperCase()}`)}</span>
						</td>

						<td
							data-testid="LinkCollectionTable__row__value"
							className={rowIndex > 0 ? "py-6" : "pb-6 pt-2"}
						>
							<input
								className="focus:outline-none"
								name={`${name}[${rowIndex}].value`}
								ref={registerRef()}
								defaultValue={rowData.value}
								readOnly
							/>
						</td>

						<td className={`w-16 text-right ${rowIndex === 0 && "pb-4"}`}>
							<Button
								data-testid="LinkCollectionTable__remove-link"
								size="icon"
								variant="plain"
								onClick={() => remove(rowIndex)}
							>
								<Icon name="Trash" />
							</Button>
						</td>
					</tr>
				)}
			</Table>
		</div>
	);
};
