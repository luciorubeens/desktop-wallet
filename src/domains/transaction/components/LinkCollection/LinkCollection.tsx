import React from "react";
import { Control, useFieldArray, ValidationRules } from "react-hook-form";

import { EntityLink } from "./LinkCollection.models";
import { LinkCollectionContainer } from "./LinkCollectionContainer";
import { LinkCollectionForm } from "./LinkCollectionForm";
import { LinkCollectionTable } from "./LinkCollectionTable";

type Props = {
	selectOptions: { label: string; value: string }[];
	itemLabel: string;
	title: string;
	description: string;
	name: string;
	registerRef: (options?: ValidationRules) => (ref: HTMLInputElement | null) => void;
	control: Control<Record<string, any>>;

	checkOptionsNames?: string[];
	checkColumnTitle?: string;
	onChecked?: (index: number) => void;
	checkedIndex?: number;
};

export const LinkCollection = ({
	selectOptions,
	title,
	description,
	name,
	registerRef,
	control,
	itemLabel,
	checkColumnTitle,
	checkOptionsNames,
	checkedIndex,
	onChecked,
}: Props) => {
	const { fields, append, remove } = useFieldArray({
		control,
		name,
		keyName: "value",
	});

	return (
		<>
			<LinkCollectionContainer title={title} description={description}>
				<LinkCollectionForm append={append} selectOptions={selectOptions} itemLabel={itemLabel} />

				{fields.length ? (
					<>
						<p
							data-testid="LinkCollection__label"
							className="mt-8 mb-2 text-sm text-theme-neutral-dark font-semibold"
						>
							Your {itemLabel}
						</p>
						<LinkCollectionTable
							remove={remove}
							fields={fields as EntityLink[]}
							name={name}
							registerRef={registerRef}
							checkColumnTitle={checkColumnTitle}
							checkOptionsNames={checkOptionsNames}
							checkedIndex={checkedIndex}
							onChecked={onChecked}
						/>
					</>
				) : (
					React.Fragment
				)}
			</LinkCollectionContainer>
		</>
	);
};
