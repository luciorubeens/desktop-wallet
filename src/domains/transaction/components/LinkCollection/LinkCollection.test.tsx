import { renderHook } from "@testing-library/react-hooks";
import React from "react";
import { useForm } from "react-hook-form";
import { render } from "utils/testing-library";

import { LinkCollection } from "./LinkCollection";

const types = [
	{
		label: "Facebook",
		value: "facebook",
	},
	{
		label: "Twitter",
		value: "twitter",
	},
	{
		label: "Instagram",
		value: "instagram",
	},
];

describe("LinkCollection", () => {
	it("should render without fields", () => {
		const { result } = renderHook(() => useForm());
		const { control, register } = result.current;

		const { asFragment, getByTestId, queryByTestId } = render(
			<LinkCollection
				title="Social Media"
				description="Tell people more about yourself through social media"
				selectOptions={types}
				name="socialMedia"
				itemLabel="media"
				registerRef={register}
				control={control}
			/>,
		);

		expect(getByTestId("LinkCollectionContainer")).toBeInTheDocument();
		expect(getByTestId("LinkCollectionForm")).toBeInTheDocument();
		expect(queryByTestId("LinkCollectionTable")).not.toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render with fields", () => {
		const { result } = renderHook(() =>
			useForm({
				defaultValues: {
					socialMedia: [
						{
							type: "facebook",
							value: "https://facebook.com",
						},
						{
							type: "instagram",
							value: "https://instagram.com",
						},
					],
				},
			}),
		);
		const { control, register } = result.current;

		const { asFragment, getByTestId, queryByTestId } = render(
			<LinkCollection
				title="Social Media"
				description="Tell people more about yourself through social media"
				selectOptions={types}
				name="socialMedia"
				itemLabel="media"
				registerRef={register}
				control={control}
			/>,
		);

		expect(getByTestId("LinkCollectionContainer")).toBeInTheDocument();
		expect(getByTestId("LinkCollectionForm")).toBeInTheDocument();
		expect(getByTestId("LinkCollection__label")).toBeInTheDocument();
		expect(getByTestId("LinkCollectionTable")).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});
});
