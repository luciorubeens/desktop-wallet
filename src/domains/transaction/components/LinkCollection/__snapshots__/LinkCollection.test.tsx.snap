// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkCollection should render with fields 1`] = `
<DocumentFragment>
  <div
    data-testid="LinkCollectionContainer"
  >
    <div
      class="flex items-center justify-between cursor-pointer"
      data-testid="LinkCollectionContainer__header"
    >
      <span
        class="text-lg font-semibold"
      >
        Social Media
      </span>
      <div
        data-testid="LinkCollectionContainer__collapse-toggle"
      >
        <div
          class="sc-AxjAm iepfoc flex items-center justify-center w-5 h-5 rounded-full text-theme-primary bg-theme-primary-contrast"
          height="10"
          width="10"
        >
          <svg>
            chevron-down.svg
          </svg>
        </div>
      </div>
    </div>
    <p
      class="mt-2 text-theme-neutral-dark"
      data-testid="LinkCollectionContainer__description"
    >
      Tell people more about yourself through social media
    </p>
    <div
      aria-hidden="true"
      data-testid="LinkCollectionContainer__collapse"
      style="height: 0px; max-height: auto; opacity: 0; overflow: hidden;"
    >
      <div
        class="mt-4"
      >
        <form
          class="grid col-gap-2 row-gap-4"
          data-testid="LinkCollectionForm"
          style="grid-template-columns: 40% 60%;"
        >
          <fieldset
            class="sc-AxhCb ckQmOL flex flex-col space-y-2"
          >
            <label
              class="flex inline-block mb-2 text-sm font-semibold transition-colors duration-100 FormLabel text-theme-neutral-dark"
              data-testid="FormLabel"
              for="type"
            >
              Add media
              <div
                class="w-1 h-1 mt-1 ml-1 rounded-full bg-theme-danger-400"
                data-testid="FormLabel__required"
              />
            </label>
            <div
              class="relative w-full"
            >
              <input
                class="sc-AxhUy jqmzti overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-neutral-300 text-theme-neutral-900 transition-colors duration-200 px-4 py-3 sr-only"
                data-testid="select-list__input"
                name="type"
                readonly=""
                type="text"
                value=""
              />
              <div
                class="relative w-full cursor-pointer"
              >
                <div>
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="downshift-1-label downshift-1-toggle-button"
                    class="sc-fzplWN jAabRP   "
                    data-testid="select-list__toggle-button"
                    id="downshift-1-toggle-button"
                    type="button"
                  />
                  <ul
                    aria-labelledby="downshift-1-label"
                    class="sc-fznyAO dPDaFu"
                    id="downshift-1-menu"
                    role="listbox"
                    tabindex="-1"
                  />
                </div>
                <div
                  class="sc-AxgMl sc-Axmtr hWILTO w-10 pointer-events-none text-theme-neutral"
                >
                  <div
                    class="sc-AxjAm dKDmBo"
                    height="8"
                    width="8"
                  >
                    <svg>
                      arrowdown.svg
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </fieldset>
          <fieldset
            class="sc-AxhCb ckQmOL flex flex-col space-y-2"
          >
            <label
              class="flex inline-block mb-2 text-sm font-semibold transition-colors duration-100 FormLabel text-theme-neutral-dark"
              data-testid="FormLabel"
              for="value"
            >
              Link
              <div
                class="w-1 h-1 mt-1 ml-1 rounded-full bg-theme-danger-400"
                data-testid="FormLabel__required"
              />
            </label>
            <input
              class="sc-AxhUy jqmzti overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-neutral-300 text-theme-neutral-900 transition-colors duration-200 px-4 py-3 "
              data-testid="LinkCollectionForm__value"
              name="value"
            />
          </fieldset>
          <button
            class="sc-AxiKw jBPYDN col-span-2"
            color="primary"
            data-testid="LinkCollectionForm__submit"
            type="submit"
          >
            Add Link
          </button>
        </form>
        <p
          class="mt-8 mb-2 text-sm text-theme-neutral-dark font-semibold"
          data-testid="LinkCollection__label"
        >
          Your media
        </p>
        <div
          data-testid="LinkCollectionTable"
        >
          <div
            class="sc-fznZeY gcZndM"
            role="table"
          >
            <table
              class="table-auto"
            >
              <thead>
                <tr
                  role="row"
                >
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--0"
                    role="columnheader"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    <div
                      class="flex flex-inline align-top "
                    >
                      <div>
                        Name
                      </div>
                      <div
                        class="flex items-center ml-2 text-theme-color-neutral"
                        data-testid="table__Sort"
                      >
                        <div
                          class="sc-AxjAm hdgaLj"
                          height="12"
                          width="15"
                        >
                          <svg>
                            sort.svg
                          </svg>
                        </div>
                      </div>
                    </div>
                  </th>
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--1"
                    role="columnheader"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    <div
                      class="flex flex-inline align-top "
                    >
                      <div>
                        Link
                      </div>
                      <div
                        class="flex items-center ml-2 text-theme-color-neutral"
                        data-testid="table__Sort"
                      >
                        <div
                          class="sc-AxjAm hdgaLj"
                          height="12"
                          width="15"
                        >
                          <svg>
                            sort.svg
                          </svg>
                        </div>
                      </div>
                    </div>
                  </th>
                  <th
                    class="text-sm text-left select-none text-theme-neutral"
                    colspan="1"
                    data-testid="table__th--2"
                    role="columnheader"
                  >
                    <div
                      class="flex flex-inline align-top "
                    >
                      <div>
                         
                      </div>
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                role="rowgroup"
              >
                <tr
                  class="font-semibold border-b border-theme-neutral-200"
                  data-testid="LinkCollectionTable__row"
                >
                  <td
                    class="w-40 pb-6 pt-2"
                    data-testid="LinkCollectionTable__row__type"
                  >
                    <input
                      class="hidden"
                      name="socialMedia[0].type"
                      readonly=""
                      value="facebook"
                    />
                    <span>
                      Facebook
                    </span>
                  </td>
                  <td
                    class="pb-6 pt-2"
                    data-testid="LinkCollectionTable__row__value"
                  >
                    <input
                      class="focus:outline-none"
                      name="socialMedia[0].value"
                      readonly=""
                      value="https://facebook.com"
                    />
                  </td>
                  <td
                    class="w-16 text-right pb-4"
                  >
                    <button
                      class="sc-AxiKw jEmgKd"
                      color="primary"
                      data-testid="LinkCollectionTable__remove-link"
                      type="button"
                    >
                      <div
                        class="sc-AxjAm cpczEI"
                        height="1em"
                        width="1em"
                      >
                        <svg>
                          trash.svg
                        </svg>
                      </div>
                    </button>
                  </td>
                </tr>
                <tr
                  class="font-semibold border-b border-theme-neutral-200"
                  data-testid="LinkCollectionTable__row"
                >
                  <td
                    class="w-40 py-6"
                    data-testid="LinkCollectionTable__row__type"
                  >
                    <input
                      class="hidden"
                      name="socialMedia[1].type"
                      readonly=""
                      value="instagram"
                    />
                    <span>
                      Instagram
                    </span>
                  </td>
                  <td
                    class="py-6"
                    data-testid="LinkCollectionTable__row__value"
                  >
                    <input
                      class="focus:outline-none"
                      name="socialMedia[1].value"
                      readonly=""
                      value="https://instagram.com"
                    />
                  </td>
                  <td
                    class="w-16 text-right false"
                  >
                    <button
                      class="sc-AxiKw jEmgKd"
                      color="primary"
                      data-testid="LinkCollectionTable__remove-link"
                      type="button"
                    >
                      <div
                        class="sc-AxjAm cpczEI"
                        height="1em"
                        width="1em"
                      >
                        <svg>
                          trash.svg
                        </svg>
                      </div>
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`LinkCollection should render without fields 1`] = `
<DocumentFragment>
  <div
    data-testid="LinkCollectionContainer"
  >
    <div
      class="flex items-center justify-between cursor-pointer"
      data-testid="LinkCollectionContainer__header"
    >
      <span
        class="text-lg font-semibold"
      >
        Social Media
      </span>
      <div
        data-testid="LinkCollectionContainer__collapse-toggle"
      >
        <div
          class="sc-AxjAm iepfoc flex items-center justify-center w-5 h-5 rounded-full text-theme-primary bg-theme-primary-contrast"
          height="10"
          width="10"
        >
          <svg>
            chevron-down.svg
          </svg>
        </div>
      </div>
    </div>
    <p
      class="mt-2 text-theme-neutral-dark"
      data-testid="LinkCollectionContainer__description"
    >
      Tell people more about yourself through social media
    </p>
    <div
      aria-hidden="true"
      data-testid="LinkCollectionContainer__collapse"
      style="height: 0px; max-height: auto; opacity: 0; overflow: hidden;"
    >
      <div
        class="mt-4"
      >
        <form
          class="grid col-gap-2 row-gap-4"
          data-testid="LinkCollectionForm"
          style="grid-template-columns: 40% 60%;"
        >
          <fieldset
            class="sc-AxhCb ckQmOL flex flex-col space-y-2"
          >
            <label
              class="flex inline-block mb-2 text-sm font-semibold transition-colors duration-100 FormLabel text-theme-neutral-dark"
              data-testid="FormLabel"
              for="type"
            >
              Add media
              <div
                class="w-1 h-1 mt-1 ml-1 rounded-full bg-theme-danger-400"
                data-testid="FormLabel__required"
              />
            </label>
            <div
              class="relative w-full"
            >
              <input
                class="sc-AxhUy jqmzti overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-neutral-300 text-theme-neutral-900 transition-colors duration-200 px-4 py-3 sr-only"
                data-testid="select-list__input"
                name="type"
                readonly=""
                type="text"
                value=""
              />
              <div
                class="relative w-full cursor-pointer"
              >
                <div>
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="downshift-0-label downshift-0-toggle-button"
                    class="sc-fzplWN jAabRP   "
                    data-testid="select-list__toggle-button"
                    id="downshift-0-toggle-button"
                    type="button"
                  />
                  <ul
                    aria-labelledby="downshift-0-label"
                    class="sc-fznyAO dPDaFu"
                    id="downshift-0-menu"
                    role="listbox"
                    tabindex="-1"
                  />
                </div>
                <div
                  class="sc-AxgMl sc-Axmtr hWILTO w-10 pointer-events-none text-theme-neutral"
                >
                  <div
                    class="sc-AxjAm dKDmBo"
                    height="8"
                    width="8"
                  >
                    <svg>
                      arrowdown.svg
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </fieldset>
          <fieldset
            class="sc-AxhCb ckQmOL flex flex-col space-y-2"
          >
            <label
              class="flex inline-block mb-2 text-sm font-semibold transition-colors duration-100 FormLabel text-theme-neutral-dark"
              data-testid="FormLabel"
              for="value"
            >
              Link
              <div
                class="w-1 h-1 mt-1 ml-1 rounded-full bg-theme-danger-400"
                data-testid="FormLabel__required"
              />
            </label>
            <input
              class="sc-AxhUy jqmzti overflow-hidden w-full bg-theme-background appearance-none rounded border border-theme-neutral-300 text-theme-neutral-900 transition-colors duration-200 px-4 py-3 "
              data-testid="LinkCollectionForm__value"
              name="value"
            />
          </fieldset>
          <button
            class="sc-AxiKw jBPYDN col-span-2"
            color="primary"
            data-testid="LinkCollectionForm__submit"
            type="submit"
          >
            Add Link
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
